### PR TITLE
multiple entanglements (again)

### PR DESCRIPTION
**Pull request by manno**
already reviewed in https://github.com/cloudfoundry-incubator/cf-operator/pull/774, but merged into the wrong  branch 

[#170668933](https://www.pivotaltracker.com/story/show/170668933)

### CODE REVIEW

**Comment by mook-as** [comment]
Hi! API question: This modifies a documented API of cf-operator 1.0; does this require more than usual amounts of process for that?

Thanks!

File: pkg/kube/apis/boshdeployment/v1alpha1/types.go [hunk 1]
```diff
@@ -34,10 +34,10 @@ var (
...
 	// LabelDeploymentSecretType is the label key for secret type
 	LabelDeploymentSecretType = fmt.Sprintf("%s/secret-type", apis.GroupName)
-	// AnnotationLinkProviderName is the annotation key for link provider name
-	AnnotationLinkProviderName = fmt.Sprintf("%s/link-provider-name", apis.GroupName)
-	// AnnotationLinkProviderType is the annotation key for link provider type
-	AnnotationLinkProviderType = fmt.Sprintf("%s/link-provider-type", apis.GroupName)
+	// AnnotationLinkProvidesKey is the key for the quarks links 'provides' JSON
+	AnnotationLinkProvidesKey = fmt.Sprintf("%s/provides", apis.GroupName)
+	// AnnotationLinkProviderService is the annotation key used on services to identify the link provider
+	AnnotationLinkProviderService = fmt.Sprintf("%s/link-provider-name", apis.GroupName)
```
Comment (edwardstudy) [inline pkg/kube/apis/boshdeployment/v1alpha1/types.go]: Could it change to `%s/link-provider-service`?

Comment (viovanov) [reply pkg/kube/apis/boshdeployment/v1alpha1/types.go]: @edwardstudy why do you think that's better?
It's the name of the provider, set on a service - it's not the name of the service.

**Comment by viovanov** [comment]
> Hi! API question: This modifies a documented API of cf-operator 1.0; does this require more than usual amounts of process for that?
> 
> Thanks!

Yes @mook-as  - next release will be `2.0.0`

**Pull request merged**